### ISSUE TITLE
Custom `Tab` layouts are not accessible

### ISSUE THREAD

**Issue by ericgio** [react-bootstrap/react-bootstrap#6393]
### Prerequisites

- [X] I am using the [correct version](https://github.com/react-bootstrap/react-bootstrap#bootstrap-compatibility) of React-Bootstrap for my version of Bootstrap
- [X] I have [searched](https://github.com/react-bootstrap/react-bootstrap/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/react-bootstrap/react-bootstrap/blob/master/CONTRIBUTING.md)

### Describe the bug

Custom `Tab` layouts no longer appear to be keyboard-accessible in v2. Tab elements are not focusable via tabbing, and the selected tab can no longer be changed using the arrow keys. This seems to be a regression from v1.x.

- [v2 example (broken)](https://react-bootstrap.netlify.app/components/tabs/#custom-tab-layout)
- [v1 example (working)](https://react-bootstrap-v4.netlify.app/components/tabs/#tabs-custom-layout)

### Expected behavior

I expect v2 to work the way v1 did:
- Individual tabs should be focusable using the tab key
- A focused, unselected tab can be selected by hitting the space or enter keys
- When the tablist is focused, the selected tab can be changed using the arrow keys

### To Reproduce

Navigate to the [v2 example](https://react-bootstrap.netlify.app/components/tabs/#custom-tab-layout) and try to access the tabs using a keyboard.

### Reproducible Example

[v2 example](https://react-bootstrap.netlify.app/components/tabs/#custom-tab-layout)

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of React-Bootstrap are you using?

2.5

### What version of Bootstrap are you using?

5.2

**Comment by kyletsang** [comment]
The tabbing functionality is actually correct according to the guidelines:
https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/

> - When focus moves into the tab list, places focus on the active tab element.
> - When the tab list contains the focus, moves focus to the next element in the page tab sequence outside the tablist, which is the tabpanel unless the first element containing meaningful content inside the tabpanel is focusable.

For the arrow key functionality, v1 used to add `href="#"` in the anchor automatically, but v2 doesn't.  I can't remember if there was a reason for this change, but you can fix the example by adding that to the `NavLink`s

**Comment by ericgio** [comment]
> For the arrow key functionality, v1 used to add href="#" in the anchor automatically, but v2 doesn't. I can't remember if there was a reason for this change, but you can fix the example by adding that to the NavLinks

Ah ok, that was the key piece missing. Without the `href`, the tablist isn't focusable at all via keyboard. If you aren't planning on auto-applying the `href` going forward, maybe consider calling out that custom tabs need to apply this themselves, since it's not very obvious. I'm happy to submit a PR updating the description if you'd like.

**Comment by kyletsang** [comment]
I took a closer look at the base anchor we use and it looks like the # is only set for trivial hrefs and if a role isn't set:
https://github.com/react-restart/ui/blob/96cc91cbedbfd1a0ad2041bf4a2d2c6b57af8296/src/Anchor.tsx#L35

We set a role of `tab`, so it skips this block.  In contrast, v1 doesn't check role.  

@jquense would it be fine to check `props.role === 'tab'`, then apply button logic?

ie.
```
if ((isTrivialHref(props.href) && !props.role) || props.role === 'button' || props.role === 'tab')
```
